packages/SwingSet/src/types-external.js

```diff
@@ -366,4 +367,45 @@ export {};
  *
  * @typedef { { enableDisavow?: boolean } } HasEnableDisavow
  * @typedef { DynamicVatOptions & HasEnableDisavow } StaticVatOptions
+ *
+ * @typedef { { vatParameters?: *, upgradeMessage: string } } VatUpgradeOptions
+ * @typedef { { incarnationNumber: number } } VatUpgradeResults
+ *
+ * @callback ShutdownWithFailure
+ * Called to shut something down because something went wrong, where the reason
+ * is supposed to be an Error that describes what went wrong. Some valid
+ * implementations of `ShutdownWithFailure` will never return, either
+ * because they throw or because they immediately shutdown the enclosing unit
+ * of computation. However, they also might return, so the caller should
+ * follow this call by their own defensive `throw reason;` if appropriate.
+ *
+ * @param {Error} reason
+ * @returns {void}
+ *
+ * @typedef {object} VatAdminFacet
```

Comment:
I find myself confusing this type as corresponding with the vatAdmin _service_... can we find a better name for it? Maybe "VatHandle" as in https://en.wikipedia.org/wiki/Handle_(computing) ? I'm not completely satisfied with that, but it does at least communicate the single-vat orientation and suggest some kind of interaction capability.

Reply:
Yeah, I've had that problem too. In conversation, I've tried to use "admin facet" to refer to the single-vat thing, and "VatAdminService" to refer to the kernel-wide launcher service. But when moving this I realized that `AdminFacet` would be used in places that lacked the "this is for a single vat" context, and might be less confusing if the word "vat" got thrown in there somewhere. Something like "management facet" might help.

I can't think of a perfect name now, and changing it everywhere would get deeper than this PR really needs to go, so I think I'll leave it for now and we can keep thinking of ways to improve it.